### ISSUE TITLE
2.2 version not working with BrcikFreedom

### ISSUE THREAD

**Issue by GreatBrickLab** [rgriebl/brickstore#216]
With Brick Freedom you are able to merge (sort) a file to add data for the parts that are already in your store.  This latest version of BrickStore is not able to open the files from BF.   The previous version 2.1 did not have this issue.

CJ

**Comment by ZZJHONS** [comment]
I tested with Bricksync and BS 2021.2.2 can open the file

**Comment by rgriebl** [comment]
Please attach such a file here and I can have a look.

**Comment by GreatBrickLab** [comment]
The merged file has the date added after the file name. When trying to open
a merged file a message pops up saying this XML document is not a
BrickStore XML file.

**Comment by rgriebl** [comment]
That doesn't help me. Please attach this file here - otherwise I can't help you.
When you reply, there's a line just below the text that says "Attach files by dragging..." ... either do that or click the line itself.

**Comment by rgriebl** [comment]
I'd really like to fix this for the next version, but I can't without a sample file.

**Comment by GreatBrickLab** [comment]
Sorry,  lost track of this issue.   Here are the before and after files
  


[mini475.zip](https://github.com/rgriebl/brickstore/files/6047173/mini475.zip)

**Comment by ZZJHONS** [comment]
I sent a message to the Brick Freedom dev about the issue, so he can fix the generation of the files too.

**Comment by rgriebl** [comment]
He generated a well-formed BrickStock file. I just tightened the XML parser correctness checks in BrickStore a bit and wouldn't accept files without a `<!DOCTYPE BrickStore>`. But this is (a) wrong (doctypes are optional according to the XML spec) and (b) counter productive, if tools are not adding the doctype.
The checks have been relaxed now.
I have also started to document the BSX format: https://github.com/rgriebl/brickstore/blob/master/BrickStoreXML.rnc
